o2a/utils/template_utils.py

```diff
@@ -18,13 +18,17 @@
 import jinja2
 
 from o2a.definitions import TPL_PATH
+from o2a.utils.el_utils import escape_string_with_python_escapes
 from o2a.utils.variable_name_utils import convert_to_python_variable
 
 TEMPLATE_LOADER = jinja2.FileSystemLoader(searchpath=TPL_PATH)
-TEMPLATE_ENV = jinja2.Environment(loader=TEMPLATE_LOADER, undefined=jinja2.StrictUndefined)
+TEMPLATE_ENV = jinja2.Environment(
+    loader=TEMPLATE_LOADER, undefined=jinja2.StrictUndefined, trim_blocks=True, lstrip_blocks=True
+)
 TEMPLATE_CACHES: Dict[str, Any] = {}
 
 TEMPLATE_ENV.filters["to_var"] = convert_to_python_variable
+TEMPLATE_ENV.filters["python_escape"] = escape_string_with_python_escapes
```

Comment:
I think, this filter should return a text with quote. This will make its use much easier. What do you think about taking this change into a separate PR?

Reply:
Yep. We could probably implement it together with implementing the generic filter for python_escape.

Reply:
I started thinking about it. I think that we should not completely implement function change at all. It seems to me that it is not necessary to introduce a different way of formatting strings in this PR. At the same time losing a lot of functionality that filter to_json provides. to_json is good enough in this case.  If we introduce the current version of this filter, it will be more difficult to restore the old functionality.  I think it is worth doing it once and well.

Let's look at how it will look. We assume that the variable ARG_TEXT contains string and the variable ARG_LIST contains list of string

1. 
```python
AwesomeStuff(
   arg_text={{ arg_text | to_json }},
   {% if arg_list %}
   arg_list={{ arg_list | tojson }},
   {% endif %}
)
```
2. 
```python
AwesomeStuff(
    arg_text={{ arg_text | to_json }},
     {% if arg_list %}arg_list=[
          {% for arg in arg_list %}
            '{{ arg | python_escape }}',
        {% endfor %}],
    {% endif %}
)
```
3. 
```python
AwesomeStuff(
   arg_text={{ arg_text | python_escape }},
   {% if arg_list %}
   arg_list={{ arg_list | python_escape }},
   {% endif %}
)
```
I believe that the introduction of new functionality should be at least equivalent to the current solution. I believe that the introduction of new functionality should be as valuable as the current solution, or more. Introducing this filter in this version in this change will bring us a lot more work in the future, which will be easier to do if this change is not included.

Reply:
I think I have found a really good solution to that.

I agree that moving '' into the escape function is a good idea. But I think using a generic filter (like tojson) is very misleading because the reader does not know the intention. I reviewed all the usages of the filter so far and it seems that we only have three cases: string, list of strings and dict of string, string. In all cases the strings are optional and can be turned to None. So far we have no bool, int nor others and we always know what type we are going to escape. Therefore I think much simpler and explicit solution is to implement three filters - python_escape_string, python_escape_list, python_escape_dictionary and use those filters as appropriate. 

I implemented it in this fixup: 4fd30e46bfda45bd80b44a75f36279caee03008c 

Let me know if @mik-laj if that resolves the discussion.

Reply:
I am also open to discuss adding a wrapper "filter" to detect the type and choose one of those three (or more in the future) but I would like to discuss it in the team as I think separate filters express the intention much better.

Reply:
I prepared my proposition at https://github.com/GoogleCloudPlatform/oozie-to-airflow/pull/224